### PR TITLE
Make blockstore an installable package

### PR DESCRIPTION
**Pull request by symbolist**
## Description

This PR pull out the following changes from https://github.com/edx/blockstore/pull/97:
* Move apps/api to apps/rest_api.
* Copy the latest code in `edx-platform/openedx/core/lib/blockstore_api` to `blockstore/apps/api`. This has been done to make the review of https://github.com/edx/blockstore/pull/97 easier since we will get a useful diff of changes made to each of the API methods.
* Makes the apps into an installable package.

It also disables the migration code which changes the MySQL character set to `utfmb4` to prevent accidentally running it against an edxapp database.

## Ticket

* [FAL-170](https://tasks.opencraft.com/browse/FAL-170)

## Test Instructions

* Checkout branch.
* In the Open edX devstack run:
```
make dev.shell.mysql57
mysql -u root
DROP DATABASE blockstore_db;
```

* Start and provision the blockstore docker container.
* Run tests with `make test`.
* Start the blockstore django server and check the API is working correctly using the Libraries UI or LabXchange devstack.

Add `'blockstore.apps.bundles.apps.BundlesConfig',` to `lms/envs/common.py::INSTALLED_APPS`.

Install the package in LMS:
```
make lms-shell
pip install -e /edx/src/blockstore
./manage.py lms migrate
```

Check the API methods can be called:
```
./manage.py lms shell
from blockstore.api import *
create_collection("Collection") # This should fail with requests.exceptions.ConnectionError
```

## TODOs

None

### CODE REVIEW

File: blockstore/apps/bundles/models.py [hunk 1]
```diff
@@ -74,7 +74,7 @@ class Collection(models.Model):
     """
     id = models.BigAutoField(primary_key=True)
     uuid = models.UUIDField(default=uuid.uuid4, unique=True, editable=False)
-    title = models.CharField(max_length=MAX_CHAR_FIELD_LENGTH, db_index=True)
+    title = models.CharField(max_length=MAX_CHAR_FIELD_LENGTH, db_index=True, db_collation='utf8mb4_general_ci')
```
Comment (symbolist) [inline blockstore/apps/bundles/models.py:77]: @bradenmacdonald Turns out changing the collation also updates the table to use the associated character set:
https://dev.mysql.com/doc/refman/8.0/en/charset-table.html

So this worked!

**Comment by symbolist** [comment]
@natabene Thanks, @bradenmacdonald is already on it!

**Comment by jvdm** [comment]
@bradenmacdonald Do you think you can tackle this one today or early tomorrow?

**Comment by bradenmacdonald** [comment]
@jvdm I'll try, I'm just in the middle of rebuilding my devstack though, so it may take a bit longer than usual.

**Comment by jvdm** [comment]
@bradenmacdonald:

> @jvdm I'll try, I'm just in the middle of rebuilding my devstack though, so it may take a bit longer than usual.

OK. Let me know if you can do it today and if I can do anything to speed up the review process. Thank you.

**Comment by bradenmacdonald** [comment]
Two of [the integration tests](https://github.com/edx/blockstore#running-integration-tests) are failing for me, but they also appear to be failing on master. We should be checking those when doing any PRs or refactoring. Are they just failing for me or for others too?

```
FAILED openedx/core/djangoapps/content_libraries/tests/test_views_lti.py::LibraryBlockLtiUrlViewTest::test_lti_url_generation - AttributeError: 'ReturnDict' object has no attribute 'usage_key'
FAILED openedx/core/djangoapps/content_libraries/tests/test_views_lti.py::LibraryBlockLtiUrlViewTest::test_block_not_found - AttributeError: 'APIClient' object has no attribute 'GET'

______________________________________________________________________________________________ LibraryBlockLtiUrlViewTest.test_lti_url_generation _______________________________________________________________________________________________

self = <openedx.core.djangoapps.content_libraries.tests.test_views_lti.LibraryBlockLtiUrlViewTest testMethod=test_lti_url_generation>

    def test_lti_url_generation(self):
        """
        Test the LTI URL generated from the block ID.
        """
    
        library = self._create_library(
            slug="libgg", title="A Test Library", description="Testing library", library_type=PROBLEM,
        )
    
        block = self._add_block_to_library(library['id'], PROBLEM, PROBLEM)
>       usage_key = str(block.usage_key)
E       AttributeError: 'ReturnDict' object has no attribute 'usage_key'

openedx/core/djangoapps/content_libraries/tests/test_views_lti.py:69: AttributeError
--------------------------------------------------------------------------------------------------------------- Captured log call ---------------------------------------------------------------------------------------------------------------
ERROR    openedx.core.djangoapps.safe_sessions.middleware:middleware.py:120 SafeCookieData BWC parse error: 's961ywd6qvqjp0levjkd5h7n49d7gwrx'.
________________________________________________________________________________________________ LibraryBlockLtiUrlViewTest.test_block_not_found ________________________________________________________________________________________________

self = <openedx.core.djangoapps.content_libraries.tests.test_views_lti.LibraryBlockLtiUrlViewTest testMethod=test_block_not_found>

    def test_block_not_found(self):
        """
        Test the LTI URL cannot be generated as the block not found.
        """
    
        self._create_library(
            slug="libgg", title="A Test Library", description="Testing library", library_type=PROBLEM,
        )
    
>       self._api("GET", '/api/libraries/v2/blocks/not-existing-key/lti/', None, expect_response=404)

openedx/core/djangoapps/content_libraries/tests/test_views_lti.py:87: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <openedx.core.djangoapps.content_libraries.tests.test_views_lti.LibraryBlockLtiUrlViewTest testMethod=test_block_not_found>, method = 'GET', url = '/api/libraries/v2/blocks/not-existing-key/lti/', data = None, expect_response = 404

    def _api(self, method, url, data, expect_response):
        """
        Call a REST API
        """
>       response = getattr(self.client, method)(url, data, format="json")
E       AttributeError: 'APIClient' object has no attribute 'GET'

openedx/core/djangoapps/content_libraries/tests/base.py:153: AttributeError
```

**Review by bradenmacdonald — APPROVED** [review]
:+1: but if you have time please fix the LTI integration test before merging.

- [x] I tested this: ran integration tests (no regressions), followed instructions for testing as an app
- [x] I read through the code
- [x] I checked for accessibility issues: n/a
- [x] Includes documentation: changelog, yes

**Comment by jvdm** [comment]
@bradenmacdonald:

> +1 but if you have time please fix the LTI integration test before merging.

I was able to reproduce the failures. The tests that are failing are part of Content Libraries at are not related to Blockstore. I will submit a separate PR to fix them.

Who can actually merge this?

CC @natabene

**Review by jvdm — APPROVED** [review]
:+1:

- [ ] I tested this: 1. Tests described in the description. 2. Blockstore Integration tests. 3. Manual testing: Run content libraries frontend app, in a devstack, created a library and added 1 block (Pool). Deleted the block. All successful.
- [ ] I read through the code
- [ ] I checked for accessibility issues
- [ ] Includes documentation
- [ ] I made sure any change in configuration variables is reflected in the corresponding client's `configuration-secure` repository.

**Comment by edx-webhook** [comment]
@ormsbee, @kdmccormick: thought you might like to know that symbolist merged this pull request.

**Comment by edx-webhook** [comment]
@symbolist 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/blockstore&entry.867055334=https://github.com/edx/blockstore/pull/135&entry.1484655318=https://github.com/symbolist&entry.752974735=2021-10-25+18:36&entry.1917517419=2021-11-05+10:47&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**